R/makeTreeSummarizedExperimentFromqiime2.R

```diff
@@ -110,37 +116,51 @@ makeTreeSummarizedExperimentFromqiime2 <- function(otu_qza,
 #'   [`ape::phylo`] object for phylogenetic tree,
 #'   [`Biostrings::DNAStringSet-class`] for representative sequences of taxa.
 #' @noRd
+#' @importFrom yaml read_yaml
+#' @importFrom utils unzip
+#' @importFrom ape read.tree
+#' @importFrom Biostrings readDNAStringSet
 #' @keywords internal
-read_qza <- function(file, temp = tempdir()) {
-    unzipped_file <- utils::unzip(file, exdir = temp)
+.read_qza <- function(file, temp = tempdir()) {
+    if (!file.exists(file)) {
+        stop(file, " does not exist", call. = FALSE)
+    }
+    if (.get_ext(file) != "qza") {
+        stop("The input `file` must be in `qza` format (qiime2 Artifact)")
```

Comment:
Better would be to include the file for, which this is a problem, since the function can get called with multiple files in one call to `makeTSEfromquiime2`

```
stop("The input '",file,"' must be in `qza` format (qiime2 Artifact)", call. = FALSE)
```